rskj-core/src/test/java/co/rsk/peg/BridgeSupportPegoutTransactionCreatedEventTest.java

```diff
@@ -0,0 +1,323 @@
+        return Stream.of(
+            Arguments.of(ActivationConfigsForTest.arrowhead600().forBlock(0)),
+            Arguments.of(ActivationConfigsForTest.lovell700().forBlock(0))
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("pegoutTransactionCreatedEventArgsProvider")
+    void test_pegoutTransactionCreatedEvent_when_pegout_batch_is_created(ActivationConfig.ForBlock activations) throws IOException {
+        // Arrange
+        List<UTXO> fedUTXOs = PegTestUtils.createUTXOs(
+            10,
+            bridgeMainnetConstants.getGenesisFederation().getAddress()
+        );
+        when(provider.getNewFederationBtcUTXOs())
+            .thenReturn(fedUTXOs);
+
+        List<Entry> pegoutRequests = PegTestUtils.createReleaseRequestQueueEntries(3);
+        when(provider.getReleaseRequestQueue())
+            .thenReturn(new ReleaseRequestQueue(pegoutRequests));
+
+        PegoutsWaitingForConfirmations pegoutsWaitingForConfirmations = provider.getPegoutsWaitingForConfirmations();
+
+        BridgeSupport bridgeSupport = new BridgeSupportBuilder()
+            .withBridgeConstants(bridgeMainnetConstants)
+            .withProvider(provider)
+            .withEventLogger(eventLogger)
+            .withActivations(activations)
+            .build();
+
+        Keccak256 pegoutCreationRskTxHash = PegTestUtils.createHash3(1);
+
+        Transaction executionRskTx = mock(Transaction.class);
+        when(executionRskTx.getHash()).thenReturn(pegoutCreationRskTxHash);
+
+        // Act
+        bridgeSupport.updateCollections(executionRskTx);
+
+        // Assertions
+
+        // Assert one pegout tx was added to pegoutsWaitingForConfirmations from the creation of a pegout batch
+        assertEquals(1, pegoutsWaitingForConfirmations.getEntries().size());
+
+        PegoutsWaitingForConfirmations.Entry pegoutEntry = pegoutsWaitingForConfirmations.getEntries().stream().findFirst().get();
+
+        BtcTransaction pegoutBatchTransaction = pegoutEntry.getBtcTransaction();
+
+        Sha256Hash btcTxHash = pegoutBatchTransaction.getHash();
+
+        List<Coin> outpointValues = extractOutpointValues(pegoutBatchTransaction);
+
+        List<Keccak256> pegoutRequestRskTxHashes = pegoutRequests.stream().map(Entry::getRskTxHash).collect(
+            Collectors.toList());
+        Coin totalTransactionAmount = pegoutRequests.stream().map(Entry::getAmount).reduce(Coin.ZERO, Coin::add);
+
+        verify(eventLogger, times(1)).logBatchPegoutCreated(btcTxHash, pegoutRequestRskTxHashes);
+        verify(eventLogger, times(1)).logReleaseBtcRequested(pegoutCreationRskTxHash.getBytes(), pegoutBatchTransaction, totalTransactionAmount);
+
+        if (activations.isActive(ConsensusRule.RSKIP428)){
+            verify(eventLogger, times(1)).logPegoutTransactionCreated(btcTxHash, outpointValues);
+        } else {
+            verify(eventLogger, never()).logPegoutTransactionCreated(any(), any());
+        }
```

Comment:
i'm not a fan of doing this inside the test.


`assertPegoutTransactionCreatedEmmittedWhenSupposedTo(activations, btcTxHash, outpointValues);`

```
private void assertPegoutTransactionCreatedEmmittedWhenSupposedTo(activations, btcTxHash, outpointValues) {
     if (!activations.isActive(ConsensusRule.RSKIP428)) {
          verify(eventLogger, never()).logPegoutTransactionCreated(any(), any());
          return;
     }
     
     verify(eventLogger, times(1)).logPegoutTransactionCreated(btcTxHash, outpointValues);
}
```

wdyt?

Reply:
I don't think this `if` is a big deal compared to the benefits of the refactoring you are proposing. It's just one line each block and it is pretty clear what's being done.

In my opinion, moving it to a method does not add many benefits in terms of improving legibility and simplifying the test.